internal/eval/evalers.go

```diff
@@ -1020,31 +1013,27 @@ func entityInOneWork(env *Env, entity types.EntityUID, parent types.EntityUID) b
 	}
 }
 
-func entityInSet(env *Env, entity types.EntityUID, parents map[types.EntityUID]struct{}) bool {
-	if _, ok := parents[entity]; ok {
+func entityInSet(env *Env, entity types.EntityUID, parents mapset.Container[types.EntityUID]) bool {
+	if parents.Contains(entity) {
 		return true
 	}
-	var known map[types.EntityUID]struct{}
+	var known mapset.MapSet[types.EntityUID]
 	var todo []types.EntityUID
 	var candidate = entity
 	for {
 		if fe, ok := env.Entities[candidate]; ok {
-			for _, k := range fe.Parents {
-				if _, ok := parents[k]; ok {
-					return true
-				}
+			if fe.Parents.Intersects(parents) {
+				return true
 			}
-			for _, k := range fe.Parents {
+			fe.Parents.Iterate(func(k types.EntityUID) bool {
```

Comment:
This is fine, but as a point of interest, I wonder if using a closure causes an allocation.